### PR TITLE
fix: [jsx] remote jsx/tsx files were compiled as js/ts

### PR DESCRIPTION
**Pull request by keroxp**
- At resolving remote files, extension are ignored and `.jsx` | `.tsx` files were compiled as `.js` and `.ts`, resulting throw compilation error.
- To resolve correctly, remote files should be checked by content-type and extension together.

### CODE REVIEW

**Comment by keroxp** [comment]
Why benchmark? @ry Would you re-run actions?

**Comment by bartlomieju** [comment]
@keroxp to rerun benchmarks just push empty commit:
```
git commit --allow-empty -m "reset CI"
```

File: cli/tests/048_remote_jsx.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+import "https://raw.githubusercontent.com/denoland/deno/master/cli/tests/047_jsx_test.jsx";
```
Comment (ry) [inline cli/tests/048_remote_jsx.ts]: Use "http://127.0.0.1/cli/tests/047_jsx_test.jsx" here and elsewhere.

File: cli/file_fetcher.rs [hunk 2]
```diff
@@ -505,12 +505,16 @@ fn map_content_type(path: &Path, content_type: Option<&str>) -> msg::MediaType {
...
+          map_js_like_extension(path, msg::MediaType::TypeScript)
+        }
         "application/javascript"
         | "text/javascript"
         | "application/ecmascript"
         | "text/ecmascript"
-        | "application/x-javascript" => msg::MediaType::JavaScript,
+        | "application/x-javascript" => {
+          map_js_like_extension(path, msg::MediaType::JavaScript)
+        }
```
Comment (ry) [inline cli/file_fetcher.rs]: I think there should be a unit test added in this file to test this change...

File: cli/file_fetcher.rs [hunk 3]
```diff
@@ -1521,6 +1521,41 @@ mod tests {
...
+      ),
+      msg::MediaType::JSX
+    );
+    assert_eq!(
+      map_content_type(
+        Path::new("foo/bar.jsx"),
+        Some("application/x-typescript")
+      ),
+      msg::MediaType::JSX
+    );
```
Comment (ry) [inline cli/file_fetcher.rs]: 👍

**Comment by keroxp** [comment]
@ry I can't know how integration tests are failing on ubuntu. There are no hints on CI console, just exiting with code 1. Those tests have passed on my macOS machine. Could someone check it on linux?

**Comment by keroxp** [comment]
Additionally, `cargo test` fails on `master` branch on my macOS (10.15)

```
test wasm ... ok
test fmt_test ... FAILED
test js_unit_tests ... FAILED

failures:

---- fmt_test stdout ----
tools/http_server.py starting...
tools/http_server.py ready
target_dir /Users/keroxp/src/deno/target/debug
target_dir /Users/keroxp/src/deno/target/debug
thread 'fmt_test' panicked at 'tools/fmt_test.py executed with failing error code
fmt (__main__.TestFmt) ... ERROR

======================================================================
ERROR: fmt (__main__.TestFmt)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tools/fmt_test.py", line 31, in test_fmt
    quiet=True)
  File "/Users/keroxp/src/deno/tools/util.py", line 99, in run_output
    sys.exit(retcode)
SystemExit: -4

----------------------------------------------------------------------
Ran 1 test in 24.631s

FAILED (errors=1)
', cli/tests/util/mod.rs:34:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

---- js_unit_tests stdout ----
thread 'js_unit_tests' panicked at 'called `Result::unwrap()` on an `Err` value: "PoisonError { inner: .. }"', src/libcore/result.rs:999:5


failures:
    fmt_test
    js_unit_tests

test result: FAILED. 92 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out
```

**Comment by bartlomieju** [comment]
@keroxp it's caused by #3122 there's already PR going to fix it

**Comment by ry** [comment]
@keroxp I've merged the v8 upgrade which should alleviate the failure you were experiencing. I think there is still an issue in this patch though...

**Pull request merged**